proto/bpfd.proto

```diff
@@ -79,23 +79,24 @@ message LoadRequest {
 };
 
 message LoadResponse {
-    string id = 1;
+    uint32 id = 1;
 }
 
 message UnloadRequest {
-    string id = 1;
+    uint32 id = 1;
 }
 
 message UnloadResponse {}
 
 message ListRequest {
     optional uint32 program_type = 1;
     optional bool bpfd_programs_only = 2;
+    optional string uuid = 3;
```

Comment:
> Why do we need this? You will only have 1 result for UUID

As of now we treat uuid as just an additional piece of metadata... so if client's mess up and add two programs with the same UUID this _could_ end up being >1 results

> , which is a Get no?

No not necessarily... I don't think we should allow GET to be called with anything other than a program ID.... Let's make "querying by label, uuid etc" Always be a list so that we don't have to deal with that complexity internally

Reply:
I think going for `list by metadata` is fine. so long as it's generic and we're not special casing uuid.
e.g I can search by any random piece of metadata that fits in a map<string, string>